beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -209,6 +222,34 @@ func (s *Service) validatePendingSlots() error {
 func (s *Service) clearPendingSlots() {
 	s.pendingQueueLock.Lock()
 	defer s.pendingQueueLock.Unlock()
-	s.slotToPendingBlocks = make(map[uint64]*ethpb.SignedBeaconBlock)
+	s.slotToPendingBlocks = make(map[uint64][]*ethpb.SignedBeaconBlock)
 	s.seenPendingBlocks = make(map[[32]byte]bool)
 }
+
+func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeaconBlock) {
+	blks, ok := s.slotToPendingBlocks[slot]
```

Comment:
Note that i did not lock here and in `insertBlockToPendingQueue`. These were all locked from parent functions. I'm not sure what is sound here